From 0d400bd6331b720c7203f3f87bed9acd54881c41 Mon Sep 17 00:00:00 2001
From: Allain Legacy <allain.legacy@windriver.com>
Date: Thu, 31 Aug 2017 13:07:50 -0500
Subject: [PATCH 083/155] avs: enabling evpn vxlan mode value

We are about to enable support for BGP backed VXLAN networks so need to
enable the "evpn" VXLAN mode value.
---
 neutron/common/constants.py            | 2 --
 neutron/plugins/wrs/agent/avs/agent.py | 3 +++
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/neutron/common/constants.py b/neutron/common/constants.py
index 6d8a796..4fc17dd 100644
--- a/neutron/common/constants.py
+++ b/neutron/common/constants.py
@@ -215,8 +215,6 @@ PROVIDERNET_GRE = 'gre'
 
 PROVIDERNET_VXLAN_DYNAMIC = 'dynamic'
 PROVIDERNET_VXLAN_STATIC = 'static'
-PROVIDERNET_VXLAN_EVPN = 'evpn'
-# TODO(alegacy): add evpn to valid list.
 PROVIDERNET_VXLAN_MODES = [PROVIDERNET_VXLAN_DYNAMIC,
                            PROVIDERNET_VXLAN_STATIC]
 
diff --git a/neutron/plugins/wrs/agent/avs/agent.py b/neutron/plugins/wrs/agent/avs/agent.py
index 14f9962..6e5d26e 100644
--- a/neutron/plugins/wrs/agent/avs/agent.py
+++ b/neutron/plugins/wrs/agent/avs/agent.py
@@ -169,6 +169,9 @@ DEVICES_ON_EXTERNAL_NETWORKS = [constants.DEVICE_OWNER_ROUTER_GW,
 
 FLOODING_ENTRY_MAC = '00:00:00:00:00:00'
 
+# These modes need L2POP assistance
+L2POP_VXLAN_MODES = [n_const.PROVIDERNET_VXLAN_STATIC]
+
 
 def _is_ipv6_link_local(address):
     ipaddr = netaddr.IPAddress(address)
-- 
2.7.4

